app/src/main/java/com/better/alarm/background/VibrationPlugin.kt

```diff
@@ -35,8 +35,11 @@ class VibrationPlugin(
         Observable.combineLatest(
                 vibratePreference,
                 targetVolume,
-                BiFunction<Boolean, TargetVolume, TargetVolume> { isEnabled, volume ->
-                  if (isEnabled) volume else TargetVolume.MUTED
+                BiFunction<String, TargetVolume, TargetVolume> { vibrate, volume ->
```

Comment:
Try testing this. From the code alone I assume that changing the value for an alarm will not have any effect on the vibration, because here you evaluate only the global setting. I think `alarm: PluginAlarmData,` should contain the data you need to decide per alarm.

Reply:
I have tested it on a physical device, its works well.

Reply:
Ok, let's walk through the code.
```
Given
The preference value is "on"
And alarm checkbox is not checked

When
Alarm fires

Then
What is expected?
What actually happens?

Reply:
I am a bit confused over here. Can you please guide me, exactly what needs to be done here.

Reply:
> what needs to be done here

You have to walk through the code as if you are Android. You can try writing comments on each like. Repeat this process 6 times for this particular function with different input parameters:

* Setting on & alarm vibrate on
* Setting on & alarm vibrate off
* Setting off & alarm vibrate on
* Setting off & alarm vibrate off
* Setting alwaysOff & alarm vibrate on
* Setting alwaysOff & alarm vibrate off